packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -227,11 +240,36 @@ class ImageCache {
     return result;
   }
 
+  /// The [ImageCacheLocation] information for the given `key`.
+  ImageCacheLocation locationForKey(Object key) => ImageCacheLocation._(
+    pending: _pendingImages.containsKey(key),
+    completed: _cache.containsKey(key),
+    live: _liveImages.containsKey(key),
+  );
+
   /// Returns whether this `key` has been previously added by [putIfAbsent].
   bool containsKey(Object key) {
     return _pendingImages[key] != null || _cache[key] != null;
   }
 
+  /// The number of live images being held by the [ImageCache]. Compare with
+  /// [ImageCache.currentSize] for completed held images.
+  int get liveImageCount => _liveImages.length;
+
+  /// The number of images being tracked as pending in the [ImageCache]. Compare
+  /// with [ImageCache.currentSize] for completedly held images.
+  int get pendingImageCount => _pendingImages.length;
+
+  /// Clears any live references to images in this cache.
```

Comment:
Maybe we should define what 'live references' are somewhere here.

Reply:
Did that here and in a couple other places now.